pkgs/misc/arm-trusted-firmware/default.nix

```diff
@@ -17,11 +17,13 @@ let
       owner = "ARM-software";
       repo = "arm-trusted-firmware";
       rev = "refs/tags/v${version}";
-      sha256 = "1gm0bn2llzfzz9bfsz11fhwxj5lxvyrq7bc13fjj033nljzxn7k8";
+      sha256 = "087pkwa6slxff0aiz3v42gww007nww97bl1p96fvvs7rr1y14gjx";
     };
 
     depsBuildBuild = [ buildPackages.stdenv.cc ];
 
+    nativeBuildInputs = [ pkgsCross.arm-embedded.stdenv.cc ];
+
```

Comment:
This is probably uneeded; `depsBuildBuild` already define a proper compiler.

I have verified, after removing the added `nativeBuildInput` that it builds for `ubootRockPro64` and that it builds and boots for `ubootRockPro64`.

```
nix-build -I nixpkgs=$PWD -E 'let plat = (import <nixpkgs> {}).pkgsCross.aarch64-multiplatform; in plat.ubootSopine'
```

Works fine on-device.

(Don't forget the `pkgsCross` up top.)

Reply:
This is in ARM trusted firmware, not U-Boot. The RockPro64 U-Boot build does not currently depend on upstream ATF, because it does not work correctly yet. I added the RK3399 ATF build for future use, which does require an `arm-embedded` compiler for building the firmware for the Cortex M0 microcontroller included in the RK3399 (I think it does power management).

Reply:
I added a comment.